packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/typings.d.ts

```diff
@@ -1,2 +1,54 @@
-declare module 'pdfjs-dist';
 declare module 'pdfjs-dist/build/pdf.worker.min.js';
+
+//
+// Declare modules and their types that is referred from PDF text layer rendering.
+// Unused properties are commented out
+//
+declare module 'pdfjs-dist/lib/web/ui_utils' {
+  export class EventBus {
+    on(eventName: string, listener: any): void;
+    off(eventName: string, listener: any): void;
+    dispatch(eventName: string, args?: any): void;
+  }
+  // export function getGlobalEventBus(): EventBus;
+}
+
+declare module 'pdfjs-dist/lib/web/text_layer_builder' {
```

Comment:
Some (all?) of these types seem to be defined by `pdfjs-dist` itself: https://github.com/mozilla/pdfjs-dist/blob/master/types/web/text_layer_builder.d.ts

Reply:
The type definition is available on or after `pdfjs-dist@2.10.377` while ours is `2.2.228`. I can we upgrade the package?

Reply:
Ah, ok. No, let's update `pdfjs-dist` in a separate PR. Also, we should be using a similar version for `@types/pdfjs-dist` as the installed `pdfjs-dist`.

Reply:
OK I update the version with `2.1.7`, which is the nearest version to the `pdfjs-dist@2.2.228`.